packages/postgress/src/MessagePickupRepositoryPg.ts

```diff
@@ -0,0 +1,616 @@
+import {
+  AddMessageOptions,
+  Agent,
+  EncryptedMessage,
+  GetAvailableMessageCountOptions,
+  injectable,
+  Logger,
+  MessagePickupEventTypes,
+  MessagePickupLiveSessionRemovedEvent,
+  MessagePickupLiveSessionSavedEvent,
+  MessagePickupRepository,
+  QueuedMessage,
+  RemoveMessagesOptions,
+  TakeFromQueueOptions,
+} from '@credo-ts/core'
+import { Pool, Client } from 'pg'
+import PGPubsub from 'pg-pubsub'
+import * as os from 'os'
+import {
+  MessageState,
+  POSTGRES_PASSWORD,
+  POSTGRES_USER,
+  POSTGRES_HOST,
+  FCM_SERVICE_BASE_URL,
+} from '../config/constants'
+import { createTableMessage, tableNameLive, createTableLive } from '../config/dbCollections'
+import { ConnectionInfo } from './interfaces'
+import { MessagePickupSession } from '@credo-ts/core/build/modules/message-pickup/MessagePickupSession'
+import axios from 'axios'
+
+@injectable()
+export class MessagePickupRepositoryPg implements MessagePickupRepository {
+  private logger?: Logger
```

Comment:
Fields like logger and database parameters (postgres host, user, pass, port, maybe DB name as well) can be part of a `PosgresMessagePickupRepositoryConfig` object that will be passed to the constructor.

Reply:
Solved!